Gems/LyShine/Code/Source/UiTextInputComponent.cpp

```diff
@@ -108,6 +108,7 @@ namespace
         utf8String.erase(left, right - left);
     }
 
+#if !defined(CARBONATED) || !defined(AZ_PLATFORM_IOS)
```

Comment:
This would be more idiomatic if expressed without the use of not operators when possible. It can be confusing when using !a || !b
Consider using:
`#if defined(CARBONATED) && defined(AZ_PLATFORM_IOS)`
instead